### PR TITLE
Firestore data modeling and query customization in Android

### LINKED ISSUE #2893: Data modeling and Query customization in Cloud Firestore with Kotlin
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Proposed title of article
Data modeling and Query customization in Cloud Firestore in Android-Kotlin

Introduction paragraph (2-3 paragraphs):

Firestore is a cloud-hosted document based database. It can be used via the appropriate native SDKs in numerous platforms (iOS, Android, and web apps). Firestore is a trustworthy database because it provides flexible data storage formats, expressive querying, offline support, better identity and access management, and synchronized real-time changes.

In this tutorial, we'll learn how to `design` a Firestore database, perform universal data `structuring/modeling` and implement complex and custom queries using the `Kotlin` programming language. 
As a Software Engineer, you need these skills to stand out in the Database management sector.

Key takeaways:

By the end of this tutorial, the reader will be expected to:

- Understand the importance of data modeling.
- Be able to design a Firestore database.
- Perform simple, complex, and custom queries.
- Effectively manage a myriad database.

References:

N/A

### PR DESCRIPTION
**Pull request by nonimdiana**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #2893

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/firestore-data-modeling-and-query-customization-in-android/index.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
+### Introduction
+The Firestore database is a modern database which is object-based and uses JSON-format to store data. In this tutorial, we will learn how to prepare data, design the database, and perform simple and complex queries using the Android platform.
+
+### Prerequisites
+You'll need the following items to follow this tutorial:
+- You must have Android Studio installed.
+- Basic knowledge of Kotlin.
+- Be familiar with the Firebase Console.
```
Comment (Ericgacoki) [inline content/articles/firestore-data-modeling-and-query-customization-in-android/index.md:8]: I think Coroutines and ViewBinding knowledge is required too. I've seen you've implemented them in your code. Kindly include them and in this list.

File: content/articles/firestore-data-modeling-and-query-customization-in-android/index.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    
+    android:padding="8dp"
+    android:layout_height="match_parent" >
```
Comment (Ericgacoki) [inline content/articles/firestore-data-modeling-and-query-customization-in-android/index.md:63]: This seems to be disarranged. Please format it properly.
> Hint: Press `Ctrl`+`Alt`+`l` to auto format your code.

File: content/articles/firestore-data-modeling-and-query-customization-in-android/index.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+#### Firestore query operators
+The following operators are available for querying the data:
+
+- `<`: Less than
+- `>=`: Greater than or equal to
+- `==`: Equal to
+- `>`: Greater than
+- `<=`: Less than or equal to
+- `array-contains`: Checks if the value is present
```
Comment (Ericgacoki) [inline content/articles/firestore-data-modeling-and-query-customization-in-android/index.md:202]: I believe there are a few operators that you missed out. List them as well.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Awesome tutorial @nonimdiana. Kindly go through the comments attached below and let me know when you're done.

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

**Review by Ericgacoki — APPROVED** [review]

**Comment by WanjaMIKE** [comment]
Well done gentlemen/lady. Nice to article. Keep up! @nonimdiana cc @Ericgacoki

**Comment by WanjaMIKE** [comment]
Merged #3730 into master

**Pull request merged**